### ISSUE TITLE
Cannot access 'browser' before initialization

### ISSUE THREAD

**Issue by BartoszKlonowski** [Lusito/mockzilla-webextension#16]
Hello!

I'm following the guide on how to use the mockzilla, but in fact if I do exactly the same as given in the [Solution artice](https://lusito.github.io/mockzilla/deep-mock.html#solution), I get the following error:
```
 FAIL  __tests__/subcomponents.test.tsx
  ● Test suite failed to run

    ReferenceError: Cannot access 'mockbrowser' before initialization

       6 |
       7 | const [mockbrowser, mockBrowser, mockBrowserNode] = deepMock<Browser>("browser", false);
    >  8 | jest.mock("webextension-polyfill", () => ({ browser: mockbrowser }));
         |                                                      ^
       9 |
      10 | function render(): ReactTestRenderer {
      11 |     const shrinkedView = TestRenderer.create(<ShrinkedView />);

      at __tests__/subcomponents.test.tsx:8:54
      at Object.<anonymous> (app/src/popup/subcomponents/ShrinkedView.tsx:3:1)
      at Object.<anonymous> (__tests__/subcomponents.test.tsx:3:1)
```

My environment is:
```
  "devDependencies": {
    ...and others,
    "@types/jest": "^28.1.7",
    "@types/react": "^18.0.17",
    "@types/react-dom": "^18.0.6",
    "@types/react-test-renderer": "^18.0.0",
    "@types/webextension-polyfill": "^0.9.0",
    ...and others,
    "jest": "^27.0.5",
    "mockzilla": "^0.14.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-test-renderer": "^18.2.0",
    "typescript": "^4.7.4",
    "webextension-polyfill": "^0.10.0",
    ...and others
  },
  "jest": {
    "verbose": true,
    "testEnvironment": "jsdom",
    "setupFilesAfterEnv": ["<rootDir>/__tests__/setupTests.ts"]
  },
```

Is there any more info on how to avoid that?
What am I missing?

**Comment by Lusito** [comment]
Hi @BartoszKlonowski , sorry for the late reply.

The link you posted is mockzilla.  Mockzilla by itself does nothing with webextensions. mockzilla-webextension already does the jest.mock call for you. Please take a look at https://lusito.github.io/mockzilla-webextension/ instead.

That being said, the link you posted is a bit out of date and I will revisit it and try to find out your issue.
I'm guessing this has to do with jest hoisting the mocks to the top of the file. I'm pretty sure though that this worked at some point. Maybe not with recent jest versions.

**Comment by Lusito** [comment]
I just tried reproducing the issue, but using just mockzilla (not mockzilla-webextensions), I can't reproduce the issue.

One error I see on your code (that is probably not related):
```
// This:
jest.mock("webextension-polyfill", () => ({ browser: mockbrowser }));
// should be:
jest.mock("webextension-polyfill", () => browser);
```

I know the wording is a bit confusing, but mockBrowser is not the mocked object, but a builder to define your expectations. In this case, browser is the mock. If you have suggestions for better names, feel free to tell me :)

The documentation has been updated to use webextension-polyfill (instead of webextension-polyfill-ts) as an example and it now contains a hint to use mockzilla-webextension instead.
